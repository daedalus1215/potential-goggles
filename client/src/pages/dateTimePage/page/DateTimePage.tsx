import React from 'react';
import { useLoaderData } from 'react-router-dom';
import { displayMsInFractionalHourFormat } from '../../../utils';
import { Task } from '../../../interfaces';
import DateTimeListView from '../dateTimeListView/DateTimeListView';
import styles from './DateTimePage.module.css';

const DateTimePage: React.FC = () => {
  const task = useLoaderData() as Task;

  if (!task) {
    throw new Response("", {
      status: 404,
      statusText: "Task not found!",
    });
  }

  const millisecondsInFractionalHourFormat = displayMsInFractionalHourFormat(task.time);
  return (<>
  
  <div className={styles.childrenContent}>
      <DateTimeListView task={task} />
    </div>
    </>
  );
};

export default DateTimePage;
