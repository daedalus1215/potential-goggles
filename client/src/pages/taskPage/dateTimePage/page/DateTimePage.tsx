import React from 'react';
import { displayMsInFractionalHourFormat } from '../../../../utils/';
import { DateTime, Task } from '../../../interfaces';
import DateTimeListView from '../dateTimeListView/DateTimeListView';
import EditDateTimeForm from '../EditDateTimeForm/EditDateTimeForm';
import styles from './DateTimePage.module.css';

interface DateTimePageProp {
  setIsShowing: (isShowing: boolean) => void;
  task: Task
}

const DateTimePage: React.FC<DateTimePageProp> = ({ task, setIsShowing }) => {
  const [editDateTime, setEditDateTime] = React.useState({ id: '', date: '', minutes: "00:00" });
  // const { dateTimes, time } = useTaskByIdSelector();
  const millisecondsInFractionalHourFormat = displayMsInFractionalHourFormat(task.time);

  return (
    <div className={styles.childrenContent}>
      {!editDateTime?.id ? (
        <>
          <DateTimeListView dateTimes={task.dateTimes} setEditDateTime={setEditDateTime} />
          <div className={styles.dateTimeTotal}> Total: {millisecondsInFractionalHourFormat} hrs</div>
        </>
      ) : (
        <EditDateTimeForm taskId={task._id} editDateTime={editDateTime} setIsShowingEditDateTimeForm={setIsShowing} />
      )}
    </div>
  );
};

export default DateTimePage;
