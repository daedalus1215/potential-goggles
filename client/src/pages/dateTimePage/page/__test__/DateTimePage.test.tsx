import React from "react";
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

// import target
import DateTimePage from '../DateTimePage';

// mock
jest.mock("react-router-dom", () => ({
    ...(jest.requireActual("react-router-dom") as any),
    useLoaderData: () => ({ _id: 'randoTaskId', title: 'randoTitle' }),
    useSubmit: () => jest.fn(),
    Form: ({ children }: any) => <div data-testid="Form">({children})</div>,
}));
jest.mock('@/components', () => ({
    TopBar: ({ children }: any) => <div data-testid="TopBar">({children})</div>,
    AddButton: () => <div data-testid="AddButton">AddButton</div>,
    BackButton: ({ path }: any) => <div data-testid="BackButton">{path}</div>
}));
jest.mock('../../dateTimeListView/DateTimeListView', () => ({
    default: ({task}: any) => <div data-testid="DateTimeListView">{task._id}</div>
}))

describe('DateTimePage.test.tsx', () => {
    it('should render...', async () => {
        // Arrange & Act
        const target = render(<DateTimePage />);

        // Assert
        expect(target.queryByTestId("BackButton")).toBeInTheDocument();
        expect(target.queryByTestId("BackButton")).toHaveTextContent('/task/randoTaskId');
        
        expect(target.queryByTestId("AddButton")).toBeInTheDocument();

        expect(target.queryByTestId("DateTimeListView")).toBeInTheDocument();
        expect(target.queryByTestId("DateTimeListView")).toHaveTextContent('randoTaskId');
        expect(target).toMatchSnapshot();

    });
});