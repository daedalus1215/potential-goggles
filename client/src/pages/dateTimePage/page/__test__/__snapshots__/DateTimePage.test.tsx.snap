// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePage.test.tsx should render... 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="header"
        >
          <h2>
            randoTitle
          </h2>
          <div
            data-testid="TopBar"
          >
            (
            <div
              data-testid="BackButton"
            >
              /task/randoTaskId
            </div>
            <div
              data-testid="Form"
            >
              (
              <input
                name="taskId"
                type="hidden"
                value="randoTaskId"
              />
              <div
                data-testid="AddButton"
              >
                AddButton
              </div>
              )
            </div>
            )
          </div>
        </div>
        <div
          class="childrenContent"
        >
          <div
            data-testid="DateTimeListView"
          >
            randoTaskId
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="header"
      >
        <h2>
          randoTitle
        </h2>
        <div
          data-testid="TopBar"
        >
          (
          <div
            data-testid="BackButton"
          >
            /task/randoTaskId
          </div>
          <div
            data-testid="Form"
          >
            (
            <input
              name="taskId"
              type="hidden"
              value="randoTaskId"
            />
            <div
              data-testid="AddButton"
            >
              AddButton
            </div>
            )
          </div>
          )
        </div>
      </div>
      <div
        class="childrenContent"
      >
        <div
          data-testid="DateTimeListView"
        >
          randoTaskId
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
